packages/@react-stately/tooltip/src/TooltipManager.ts

```diff
@@ -10,67 +10,76 @@
  * governing permissions and limitations under the License.
  */
 
-const  TOOLTIP_DELAY = 220; // --spectrum-global-animation-duration-400
+import {MultipleTolltipManager, TooltipState} from './types';
 
-export class TooltipManager {
+const CLOSE_DELAY = 220; // --spectrum-global-animation-duration-400
+const WARMUP_DELAY = 350;  // --spectrum-global-animation-duration-700
+const COOLDOWN_DELAY = 400; // --spectrum-global-animation-duration-1000
 
-  visibleTooltip?: null | {triggerId: string, state: {open: boolean, setOpen(value: boolean), tooltipManager: string}}
-  hoverHideTimeout?: null | ReturnType<typeof setTimeout>;
-  hoverShowTimeout?: null | ReturnType<typeof setTimeout>;
+let visibleTooltip: null | TooltipState = null;
```

Comment:
you should be able to keep these inside the class TooltipManager

then, to make the entire class a protected singleton you can do this

```
class TooltipManager implements MultipleTolltipManager {...}
let tooltipManager = new TooltipManager();
export {tooltipManager as TooltipManager}
```


Another approach you could take is to leave these out as they are, get rid of the class entirely and follow a reducer pattern
https://github.com/adobe-private/react-spectrum-v3/blob/398de48a8b5d653f699f0a993df42734fbe2c9a8/packages/%40react-stately/data/src/useAsyncList.ts#L79

Reply:
need little help with this

1. if TooltipManager is singleton then there is a problem with passing state correctly. I'll need to pass state in useTooltipTrigger eveytime when TooltipManager is called

something like:
```
 function useTooltipTrigger(props, state) {
  ...
  let {pressProps} = usePress({
    onPress: () => state.tooltipManager.closeTooltip(state)
  });
}
```
would this be OK?

Reply:
2. I got lost with `reducer`, is there a proper way to handle timeout inside of `reducer`?
```
    warmupTimer = setTimeout(() => {
      this.openTooltip(); <- returns state {isOpen: true}
      warmedup = true;
    }, WARMUP_DELAY);
```